src/components/AppStepper/VersionStep/VersionSelector.js

```diff
@@ -21,21 +21,36 @@ function VersionSelector(props) {
         {items.map((item) => (
           <MenuItem key={item.name || item} value={item.value || item}>
             {name !== 'MY_LANGUAGE' && (item.name || item)}
-            {name === 'MY_LANGUAGE' && <FormattedMessage id={item.name} />}
+            {name === 'MY_LANGUAGE' && (
+              <FormattedMessage
+                id={item.name}
+                values={{
+                  flag: item.flag,
```

Comment:
I will be fast because I have been awake since yesterday morning (i was in the  night shift at work)

- The conditional classes are the winners for me.  Simple, elegant.
- The globe icone moving far/close to the right while you choose longer/shorter UI language.
- For flags I found this: https://prinsfrank.nl/2021/01/25/Non-existing-flag-emojis-on-windows search for NotoColorEmojiLimited at the bottom of the page. Thats an opensource font by google (https://github.com/googlefonts/noto-emoji/). If thats not on the fonts.googleapis.com, we can serve it from public folder. Your `languageAvatar` looks good, but i think colored flags is more clearer for your eyes to select faster what you want. And now I know why UI/UX designers hate flags to choose language: countries have flags, not languages. There is 2 NL flag/avatar for Dutch and Frysk, for English why UK and not USA/Australia/New Zealand etc... :)

One more thing is the tasmota language autoselect by browser locale. Now users can select their preferred language, but tasmota language choosed by browser locale or change that with UI (if applicable)?

Reply:
"Real" flag is better idea then letters or Avatar. It will look awesome I can say!
With a moving globe depending on length of language name ... I don't have any good solution.

Yes, I think that the language selected for Tasmota should stay in sync with the language selected for TasmoCompiler GUI.

Reply:
With some changes, I think looks good and acceptable:

![lang](https://user-images.githubusercontent.com/52567253/118650339-9d097000-b7e4-11eb-89aa-2a904176867c.png)

`TopAppBar.js`
```diff
@@ -55,18 +55,19 @@ class TopAppBar extends Component {
                color="inherit"
                className={classes.language}
              >
-               <LanguageIcon className={classes.leftIcon} />
                <FormattedMessage id="appBarLang" />
+               <LanguageIcon className={classes.rightIcon} />
              </Typography>
            </div>
            <Menu
+             className={classes.languageList}
              id="langs-menu"
              anchorEl={anchorEl}
              open={Boolean(anchorEl)}
              onClose={() => this.handleClose()}
              getContentAnchorEl={null}
-             anchorOrigin={{vertical: "bottom", horizontal: "center"}}
-             transformOrigin={{vertical: "top", horizontal: "center"}}
+             anchorOrigin={{vertical: "bottom", horizontal: "right"}}
+             transformOrigin={{vertical: "top", horizontal: "right"}}
            >
              {Object.keys(allMessages)
                .sort()
@ @@
```
`styles.js`
```diff
@ -10,6 +10,9 @@ const styles = (theme) => ({
    alignItems: 'center',
    cursor: 'pointer',
  },
+ languageList: {
+   marginTop: theme.spacing.unit,
+ },
  root: {
    width: '100%',
  },
@ @@
```

I'm going to test that flag fonts I linked.

Reply:
There is more space needed between flag and language name.

Reply:
The font is almost 10MB, I think thats a no-go way :)

[Here](https://www.phoca.cz/cssflags/) is a CSS+SVG flag solution. Poland flag is only 360byte, UK is 816. I'm going this way, and add more space beetween flag and language name. At the moment there is only 1 space.

Reply:
Looks very nice. I need few days to review all changes.